### DISCUSSION TITLE
Is it possible to use this project in existing baremetal k8s cluster.

### DISCUSSION THREAD

**Discussion by dm3ch** [syself/cluster-api-provider-hetzner#792] [Q&A]
I have an existing k8s cluster on bare metal hosts. It have a separate 3 hosts for k8s API and was set up using kubeadm.

Is it possible to use this project in existing k8s cluster, just to manage new worker nodes (re-setup existing workers) and keep k8s API as is?

Is it possible to somehow migrate existing cluster control-plane nodes gradually under the controller?

**Comment by batistein** [comment]
When it comes to managing clusters, different approaches need to be taken depending on the provider used. For example, while an owning process is possible with cluster-api-provider-openstack, it is currently not possible with cluster-api-provider-hetzner. However, migration is still possible, albeit with some downtime.

To migrate your cluster, start by creating a new cluster with one control-plane. Then, take an etcd snapshot from your existing cluster and restore it in the new cluster. This will result in the same k8s-api. Once this migration is complete, scale up the new cluster and scale down the old one. If you are not using any volumes, the migration is already finished. However, if you have volumes, they will need to be adopted.

In conclusion, although an owning process cannot be done with cluster-api-provider-hetzner, migration is still a viable option. By following the steps outlined above, you can move your workloads to a new cluster with minimal disruption. At Syself, we have successfully executed this process in the past.
